### PR TITLE
CEDS-6196: Standardise back links

### PR DESCRIPTION
**Pull request by murthybathini**

### CODE REVIEW

**Review by glidester — CHANGES_REQUESTED** [review]
Just one place left where we can remove the referer url

File: app/controllers/MrnEntryController.scala [hunk 1]
```diff
@@ -59,33 +56,19 @@ class MrnEntryController @Inject() (
...
 
-  def autoFill(mrn: String, maybeRefererUrl: Option[RedirectUrl] = None): Action[AnyContent] =
+  def autoFill(mrn: String): Action[AnyContent] =
     (authenticate andThen verifiedEmail andThen getData).async { implicit req =>
-      val updatedAnswers = decodeRefererUrl(maybeRefererUrl)
-        .map(decodedBackLink => req.userAnswers.copy(mrnPageRefererUrl = Some(decodedBackLink)))
-        .getOrElse(req.userAnswers)
-
-      MRN(mrn)
-        .map(updateUserAnswersAndRedirect(_, updatedAnswers))
```
Comment (glidester) [inline app/controllers/MrnEntryController.scala:87]: You are no-longer calling `updateUserAnswersAndRedirect` so the `mrn` value is not being saved in the cache any more.

We don't need the `mrnPageRefererUrl` saved anymore but we certainly do still need the `mrn` that is passed in saved to the cache!

File: test/controllers/MrnEntryControllerSpec.scala [hunk 2]
```diff
@@ -176,14 +175,6 @@ class MrnEntryControllerSpec extends ControllerSpecBase {
 
     "provided with correct MRN" should {
 
-      "call AnswersService" in {
-        controller.autoFill(mrn)(fakeRequest).futureValue
-
-        val upsertedUserAnswers = theSavedFileUploadAnswers
-        upsertedUserAnswers.mrn mustBe defined
-        upsertedUserAnswers.mrn.get mustBe MRN(mrn).get
-      }
```
Comment (glidester) [inline test/controllers/MrnEntryControllerSpec.scala:191]: This test is still valid and should be reinstated

**Review by glidester — CHANGES_REQUESTED** [review]
Need to reinstate persisting the `mrn` value

File: test/controllers/MrnEntryControllerSpec.scala [hunk 3]
```diff
@@ -184,4 +192,17 @@ class MrnEntryControllerSpec extends ControllerSpecBase {
...
+    "not clear the mrnPageRefererUrl value in the cache" when {
+      "no value is sent in the request" in {
+        val controller = mrnEntryController(validAnswers.copy(mrnPageRefererUrl = Some(validRefererUrl)))
+        controller.autoFill(mrn)(fakeRequest).futureValue
+
+        theSavedFileUploadAnswers.mrnPageRefererUrl mustBe Some(validRefererUrl)
+      }
+    }
+
+  }
```
Comment (glidester) [inline test/controllers/MrnEntryControllerSpec.scala:206]: We don't need to reinstate this test as this is regarding the `mrnPageRefererUrl` parameter that no longer exists.

**Review by glidester — CHANGES_REQUESTED** [review]

**Review by glidester — DISMISSED** [review]

**Review by IanWood-HMRC — APPROVED** [review]

**Review by IanWood-HMRC — APPROVED** [review]

**Pull request merged**